README.md

```diff
@@ -22,26 +41,183 @@ allprojects {
 }
 ```
 
-2. Add your dependency to your app's `build.gradle` file:
+### Available Artifiacts
+
+2. Add relevant dependencies to your app's `build.gradle` file:
 
 ```groovy
-implementation 'io.customer.android:sdk:<version-here>'
+implementation 'io.customer.android:tracking:<version-here>'
+implementation 'io.customer.android:messaging-push-fcm:<version-here>'
 ```
 
-Replace `version-here` with the the latest version: ![Maven Central](https://maven-badges.herokuapp.com/maven-central/io.customer.android/sdk/badge.svg)
+Replace `version-here` with the the latest version: ![Maven Central](https://maven-badges.herokuapp.com/maven-central/io.customer.android/tracking/badge.svg)
 
-# Documentation
+# Initialize the SDK
 
-...Not yet ready. Under development. 
+Before you can use the Customer.io SDK, you need to initialize it. `CustomerIO` is a singleton: you'll create it once and re-use it across your application.
 
-# Contributing
+```kotlin
+class App : Application() {
+   override fun onCreate() {
+       super.onCreate()
+      val customerIO = CustomerIO.Builder(
```

Comment:
Is there a reason for holding a reference to the `.build()` result? (`val customerIO`)

If so, it may be a good idea to explain that with a code comment in this code sample.

Reply:
🤔 So, `customerIO` is an instance of `CustomerIO` where you can use it to call public SDK functions of the SDK instead of using the singleton `CustomerIO`? 

If that is the purpose of `customerIO`, I am confused on the use case for that since the README document says "CustomerIO is a singleton: you'll create it once and re-use it across your application.". This sentence makes me think that using the singleton is the only way of using the SDK. 

I like the idea of being able to use the SDK with a singleton or not a singleton (the iOS SDK does this). If this works, I think the readme needs to be updated to not convey an assumption that singleton is the only way of using the SDK.

Reply:
Okay so let me explain it a bit more, `CustomerIO` is indeed a `singleton ` class. And to initialize it we are using `Builder Pattern`.  And `build` method of the `Builder pattern` always returns the class object and the next time `CustomerIO` instance is accessed it's always the one you created before. 

``` Koltin
class CustomerIO internal constructor(
    val config: CustomerIOConfig,
    private val api: CustomerIoApi,
) {
    companion object {
        private var instance: CustomerIO? = null

        @JvmStatic
        fun instance(): CustomerIO {
            return instance
                ?: throw IllegalStateException("CustomerIo.Builder::build() must be called before obtaining CustomerIo instance")
        }
}
```

``` Koltin
// build method of the Builder class inside 
 fun build(): CustomerIO {

            if (apiKey.isEmpty()) {
                throw IllegalStateException("apiKey is not defined in " + this::class.java.simpleName)
            }

            if (siteId.isEmpty()) {
                throw IllegalStateException("siteId is not defined in " + this::class.java.simpleName)
            }

            val config = CustomerIOConfig(
                siteId = siteId,
                apiKey = apiKey,
                region = region,
                timeout = timeout
            )

            val customerIoComponent =
                CustomerIOComponent(customerIOConfig = config, context = appContext)

            val client = CustomerIO(config, customerIoComponent.buildApi())
            instance = client
            return client
        }
```

So as you can see `instance = client`, once the instance is created it will get reused when accessed via `CustomerIO.instance()`

Reply:
I understand how the code `CustomerIO.Builder(...).build()` works. Sorry, I believe that my comments were not asking about how it works but more about the docs and use cases around it. Mind reading my comments again?

Reply:
hmm, I am not sure, what would be another way to phrase it because we wouldn't want the client to make a new instance each time with `Builder` and then use it. 

Maybe something like this,
`Before you can use the Customer.io SDK, you need to initialize it. `CustomerIO` is a singleton: once created it will be re-used until you decide to reconfigure and recreate the instance.`